### PR TITLE
Fix mbed-tool support

### LINKED ISSUE #25: CMake build fails with `"STREQUAL" "small" Unknown arguments specified`
### Description of defect

```
[EXAMPLES]> INFO     In folder 'mbed-os-example-for-google-iot-cloud'
[EXAMPLES]> INFO     Compiling mbed-os-example-for-google-iot-cloud DISCO_L475VG_IOT01A GCC_ARM
[EXAMPLES]> INFO     Executing command 'mbed-tools compile -t GCC_ARM -m DISCO_L475VG_IOT01A -c'...

#### STDOUT ####
Configuring project and generating build system...
-- The C compiler identification is GNU 7.4.0
-- The CXX compiler identification is GNU 7.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python3: /usr/bin/python3.7 (found version "3.7.4") found components: Interpreter 
-- Checking for Python package prettytable -- found
-- Checking for Python package future -- found
-- Checking for Python package jinja2 -- found
-- Checking for Python package intelhex -- found
-- Configuring incomplete, errors occurred!
See also "/builds/workspace/mbed-os-ci_cmake-cloud-example-GCC_ARM-dev/cloud-examples/mbed-os-example-for-google-iot-cloud/cmake_build/DISCO_L475VG_IOT01A/develop/GCC_ARM/CMakeFiles/CMakeOutput.log".

#### STDERR ####
CMake Error at mbed-os/tools/cmake/app.cmake:10 (include):
  include could not find load file:

    /builds/workspace/mbed-os-ci_cmake-cloud-example-GCC_ARM-dev/cloud-examples/mbed-os-example-for-google-iot-cloud/.mbedbuild/mbed_config.cmake
Call Stack (most recent call first):
  CMakeLists.txt:10 (include)


CMake Error at mbed-os/tools/cmake/toolchain.cmake:103 (include):
  include could not find load file:

    /builds/workspace/mbed-os-ci_cmake-cloud-example-GCC_ARM-dev/cloud-examples/mbed-os-example-for-google-iot-cloud/mbed-os/tools/cmake/toolchains/.cmake
Call Stack (most recent call first):
  mbed-os/tools/cmake/app.cmake:15 (include)
  CMakeLists.txt:10 (include)


CMake Error at mbed-os/tools/cmake/toolchain.cmake:104 (include):
  include could not find load file:

    /builds/workspace/mbed-os-ci_cmake-cloud-example-GCC_ARM-dev/cloud-examples/mbed-os-example-for-google-iot-cloud/mbed-os/tools/cmake/cores/.cmake
Call Stack (most recent call first):
  mbed-os/tools/cmake/app.cmake:15 (include)
  CMakeLists.txt:10 (include)


CMake Error at mbed-os/CMakeLists.txt:8 (include):
  include could not find load file:

    /builds/workspace/mbed-os-ci_cmake-cloud-example-GCC_ARM-dev/cloud-examples/mbed-os-example-for-google-iot-cloud/.mbedbuild/mbed_config.cmake


CMake Error at mbed-os/CMakeLists.txt:29 (if):
  if given arguments:

    "STREQUAL" "small"

  Unknown arguments specified


ERROR: CMake invocation failed!

More information may be available by using the command line option '-v'.

#### End of STDOUT/STDERR ####
```

#### Target(s) affected by this defect ?
DISCO_L475VG_IOT01A

#### Toolchain(s) (name and version) displaying this defect ?
GCC_ARM

#### What version of Mbed-os are you using (tag or sha) ?

9bbc8c914c7e 

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)
Mbed CLI 2

#### How is this defect reproduced ?
`mbed-tools compile -t GCC_ARM -m DISCO_L475VG_IOT01A -c`

### PR DESCRIPTION
**Pull request by harmut01**
Fixes #25

`.travis.yml`:
* Update mbed-tools commands

`drivers/COMPONENT_wifi_ism43362.lib`:
* Use the latest version to bring in compilation fixes

`CMakeLists.txt`:
* Update the configuration path generated by mbed-tools
* Remove mbed_set_mbed_target_linker_script() call which is no longer
  present in Mbed OS.

### CODE REVIEW

**Review by LDong-Arm — CHANGES_REQUESTED** [review]
Please change the base branch to "development" - click on "Edit" near the PR's title.
Would also be good to update README.md with instructions for Mbed CLI 2. Could use https://github.com/ARMmbed/mbed-os-example-blinky as a template.

Looks good otherwise, thanks.

**Review by 0xc0170 — APPROVED** [review]
Once development is target branch set

**Comment by harmut01** [comment]
Cheers @LDong-Arm, I've updated the documentation and the base branch.

**Review by LDong-Arm — APPROVED** [review]
LGTM

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,8 +21,6 @@ add_executable(${APP_TARGET})
 
 mbed_configure_app_target(${APP_TARGET})
```
Comment (LDong-Arm) [inline CMakeLists.txt:23]: @harmut01 Sorry I missed this earlier - this can be removed too, though it makes no difference to have it.

**Pull request merged**